data/windows.yaml

```diff
@@ -331,6 +385,39 @@ urls:
 - 'https://msdn.microsoft.com/en-us/library/windows/desktop/ms694515(v=vs.85).aspx'
 - 'https://msdn.microsoft.com/en-us/library/windows/desktop/ms686595(v=vs.85).aspx'
 ---
+name: WindowsCOMProperties
+doc: |
+  Various properties of Windows COM Objects.
+
+  These artifacts are meant to highlight properties of COM objects that,
+  although legitimate, are known to be associated with persistence techniques
+  or other capabilities that malware can leverage.
+
+  ShellFolder\HideOnDesktop, ShellFolder\Attributes (specifically with value
+  0xf090013d), and InprocServer\LoadWithoutCOM are associated with a technique
+  to cause iexplore or explorer to load a malicious DLL by registering a COM
+  object and invoking it through the use of Junction Folders.
+sources:
+- type: REGISTRY_VALUE
+  attributes:
+    key_value_pairs:
+    - {key: 'HKEY_LOCAL_MACHINE\Software\Classes\CLSID\*\ShellFolder', value: 'Attributes'}
+    - {key: 'HKEY_USERS\%%users.sid%%\Software\Classes\CLSID\*\ShellFolder', value: 'Attributes'}
+    - {key: 'HKEY_LOCAL_MACHINE\Software\Classes\Wow6432Node\CLSID\*\ShellFolder', value: 'Attributes'}
+    - {key: 'HKEY_USERS\%%users.sid%%\Software\Classes\Wow6432Node\CLSID\*\ShellFolder', value: 'Attributes'}
+    - {key: 'HKEY_LOCAL_MACHINE\Software\Classes\CLSID\*\ShellFolder', value: 'HideOnDesktop'}
+    - {key: 'HKEY_USERS\%%users.sid%%\Software\Classes\CLSID\*\ShellFolder', value: 'HideOnDesktop'}
+    - {key: 'HKEY_LOCAL_MACHINE\Software\Classes\Wow6432Node\CLSID\*\ShellFolder', value: 'HideOnDesktop'}
+    - {key: 'HKEY_USERS\%%users.sid%%\Software\Classes\Wow6432Node\CLSID\*\ShellFolder', value: 'HideOnDesktop'}
+    - {key: 'HKEY_LOCAL_MACHINE\Software\Classes\CLSID\*\InprocServer32', value: 'LoadWithoutCOM'}
```

Comment:
Some of these keys already exist elsewhere, such as WindowsCOMInprocServers, please combine keys where this is appropriate

Reply:
The InprocServer32 keys could be moved under WindowsCOMInprocServers, but I'm not sure whether it'd make sense to move the ShellFolder keys under there (since InprocServer32 and ShellFolder are at the same hierarchy, I wonder if you can have a GUID where ShellFolder exists alongside InProcHandler32 or LocalServer32, for instance).  I figured it'd be better to keep them all together in a new artifact, since they are used together by the one technique.  Do you think it'd be better to split them up, though, keeping the ShellFolder ones as their own key?

Reply:
If they belong together let's keep it as as-is for now. We can always change them later. The guideline / rule of thumb is roughly to keep a definition specific to a single piece of information / data, and combine them with an artifact group if there is a logical relationship between the definitions.